consensus_test.go

```diff
@@ -208,6 +129,81 @@ func TestTransition_AcceptState_Validator_VerifyFails(t *testing.T) {
 	})
 }
 
+// Test that if build proposal fails, state machine will change state from AcceptState to RoundChangeState.
+func TestTransition_AcceptState_Proposer_FailedBuildProposal(t *testing.T) {
+	buildProposalFunc := func() (*Proposal, error) {
+		return nil, errors.New("failed to build a proposal")
+	}
+
+	validatorIds := []string{"A", "B", "C"}
+	backend := newMockBackend(validatorIds, nil).HookBuildProposalHandler(buildProposalFunc)
+
+	m := newMockPbft(t, validatorIds, "A", backend)
+	m.state.view = ViewMsg(1, 0)
+	m.setState(AcceptState)
+
+	// Prepare messages
+	m.emitMsg(&MessageReq{
+		From: "A",
+		Type: MessageReq_Prepare,
+		View: ViewMsg(1, 0),
+	})
+	m.emitMsg(&MessageReq{
+		From: "B",
+		Type: MessageReq_Prepare,
+		View: ViewMsg(1, 0),
+	})
+	m.emitMsg(&MessageReq{
+		From: "C",
+		Type: MessageReq_Prepare,
+		View: ViewMsg(1, 0),
+	})
+
+	go func() {
```

Comment:
What is happening here?

Reply:
I needed cancellation when in some previous iterations I had invoked Run function. If I haven't had cancellation, it would hang within the state machine loop until timeout at least or maybe even forever.
With runCycle function invocation, we are "safe", as it will execute only single cycle.